sentry.gradle

```diff
@@ -51,132 +51,139 @@ gradle.projectsEvaluated {
         //     .findAll{!['class', 'active'].contains(it.key)}
         //     .join('\n')
 
-        def currentVariants = extractCurrentVariants(bundleTask, releases)
-        if (currentVariants == null) return
-
-        def variant = null
-        def releaseName = null
-        def versionCode = null
-        def previousCliTask = null
-
-        def nameCleanup = "${bundleTask.name}_SentryUploadCleanUp"
-        // Upload the source map several times if necessary: once for each release and versionCode.
-        currentVariants.each { key, currentVariant ->
-          variant = currentVariant[0]
-          releaseName = currentVariant[1]
-          versionCode = currentVariant[2]
-
-          try {
-            if (versionCode instanceof String) {
-                versionCode = Integer.parseInt(versionCode)
-                versionCode = Math.abs(versionCode)
+        File bundleOutputFile = new File(bundleOutput)
+        File sourcemapOutputFile = new File(sourcemapOutput)
+
+        def bofExists = bundleOutputFile.exists()
+        def sofExists = sourcemapOutputFile.exists()
+        if (bofExists && sofExists){
```

Comment:
I'm not sure this is fine, it could be that when this is executed, the files don't even exist yet.
The other thing is, the problem here is why the files don't exist and sentry-cli fails, failing silently is even worse.
So why don't we take a step back and investigate why sentry-cli fails, if the files are missing, why are they missing?

Reply:
Thanks @marandaneto .

If I'm following correctly, it's like a race condition; where the tasks are queued for sentry-cli to execute but not all the files are created yet.

So, instead of adding the check here in the gradle file, would it be better for the sentry-cli to verify if the paths that are provided to it are valid?

Reply:
No, I don't think the sentry-cli is the one responsible for this, sentry-cli should not even be called if the paths are not valid/exist.
We need to find out your use case and understand if the Gradle plugin here isn't doing its thing correctly (because of RN breaking changes or something) or if the RN engine itself is having issues.
This issue just avoids the problem, but it does not solve it.
So can yo elaborate in your use case? how to reproduce it? etc

Reply:
Sure.  

So, we have a project that the [mattermost-mobile](https://github.com/mattermost/mattermost-mobile) that we build on CircleCI.  Each build was failing and the below error was logged.

```
> Task :app:bundleReleaseJsAndAssets_SentryUpload_*********************@2.0.0+430_430 FAILED
Caching disabled for task ':app:bundleReleaseJsAndAssets_SentryUpload_*********************@2.0.0+430_430' because:
  Build cache is disabled
Task ':app:bundleReleaseJsAndAssets_SentryUpload_*********************@2.0.0+430_430' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Starting process 'command '/home/circleci/*****************/node_modules/@sentry/cli/bin/sentry-cli''. Working directory: /home/circleci/***************** Command: /home/circleci/*****************/node_modules/@sentry/cli/bin/sentry-cli --log-level error react-native gradle --bundle /home/circleci/*****************/android/app/build/generated/assets/react/release/index.android.bundle --sourcemap /home/circleci/*****************/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map --release *********************@2.0.0+430 --dist 430
Successfully started process 'command '/home/circleci/*****************/node_modules/@sentry/cli/bin/sentry-cli''
Error: spawn /home/circleci/*****************/node_modules/@sentry/cli/sentry-cli ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn /home/circleci/*****************/node_modules/@sentry/cli/sentry-cli',
  path: '/home/circleci/*****************/node_modules/@sentry/cli/sentry-cli',
  spawnargs: [
    '--log-level',
    'error',
    'react-native',
    'gradle',
    '--bundle',
    '/home/circleci/*****************/android/app/build/generated/assets/react/release/index.android.bundle',
    '--sourcemap',
    '/home/circleci/*****************/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map',
    '--release',
    '*********************@2.0.0+430',
    '--dist',
    '430'
  ]
}
:app:bundleReleaseJsAndAssets_SentryUpload_*********************@2.0.0+430_430 (Thread[Execution worker,5,main]) completed. Took 0.095 secs.

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:bundleReleaseJsAndAssets_SentryUpload_*********************@2.0.0+430_430'.
> Process 'command '/home/circleci/*****************/node_modules/@sentry/cli/bin/sentry-cli'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

Execution optimizations have been disabled for 2 invalid unit(s) of work during this build to ensure correctness.
Please consult deprecation warnings for more details.

BUILD FAILED in 9m 1s
615 actionable tasks: 610 executed, 5 up-to-date
Some of the file system contents retained in the virtual file system are on file systems that Gradle doesn't support watching. The relevant state was discarded to ensure changes to these locations are properly detected. You can override this by explicitly enabling file system watching.
\e[0m

Exited with code exit status 1

CircleCI received exit code 
```

I SSH into the machine and indeed found the files being present.  While reading the sentry.gradle file, I noticed that it would loop for each build variant and create a cliTask for it.  Then, I thought, that it's better to create that task if we have 'real' files at the end of those destination.  This error is misleading in the sense that for the 'release' build, the file exist but for 'debug' variant, there's no file being created and from that, I devised this solution.

Reply:
@avinashlng1080 something is strange then, because this task does not run for `Debug` builds.
https://github.com/getsentry/sentry-react-native/blob/891e14de0f95b100b0c71f391dde4a85adf40f37/sentry.gradle#L37

`!task.name.contains("Debug")`

Reply:
Yes @marandaneto I said `debug` as in a figure of speech. I was referring to other tasks that may have slipped into this if condition.

Reply:
Did it possibly start happening after an RN version upgrade?

Reply:
we are using `"react-native": "0.70.3",`

Reply:
It turns out that the 'sentry-cli' was not being properly installed by CircleCi.  Thanks to my team mate Elias, he ran the 'install-script' from within `node_modules/@sentry/cli/scripts/install.js` as part of a job and boom.  It all worked out correctly.